torch/csrc/autograd/engine.h

```diff
@@ -53,6 +53,11 @@ struct GraphTask {
   std::unordered_map<Node*, InputBuffer> not_ready_;
   std::unordered_map<Node*, int> dependencies_;
 
+  struct GraphTaskFunctionPreHook {
```

Comment:
nit: Should we call this GradCapturePostHook as we mentioned it in the design doc?

Reply:
Ok, I'll rename it.

Reply:
See the reply here:
https://github.com/pytorch/pytorch/pull/34501/files#r397583014